gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -68,18 +71,54 @@ public async Task ConnectAsync()
 
         public async Task CloseAsync()
         {
+            if (!String.IsNullOrEmpty(_sessionId))
```

Comment:
(nitpick) **s**tring (it's just the usual C# code style)

Reply:
One other idea I just got, how about adding a private property like this:

```c#
private bool IsSessionEnabled => !string.IsNullOrEmpty(_sessionId);
```

then this if clause could be changed to:

```c#
if (IsSessionEnabled)
```

which is easer to understand in my opinion.
But this if course only a suggestion. Ignore it if you disagree.